File: Chroma-Core/src/main/java/buttondevteam/lib/architecture/ReadOnlyConfigData.java
package buttondevteam.lib.architecture;

import java.util.function.Function;

public class ReadOnlyConfigData<T> extends ConfigData<T> {
	ReadOnlyConfigData(IHaveConfig config, String path, T def, Object primitiveDef, Function<Object, T> getter, Function<T, Object> setter) {
		super(config, path, def, primitiveDef, getter, setter);
	}

	ReadOnlyConfigData(IHaveConfig config, String path, T def, Object primitiveDef) {
		super(config, path, def, primitiveDef, null, null);
	}
}

File: Chroma-Core/src/main/java/buttondevteam/lib/architecture/ReadOnlyConfigData.kt
package buttondevteam.lib.architecture

import java.util.function.Function

class ReadOnlyConfigData<T> internal constructor(
    config: IHaveConfig?,
    path: String,
    def: T?,
    primitiveDef: Any?,
    getter: Function<Any?, T>,
    setter: Function<T, Any?>
) : ConfigData<T>(config, path, def, primitiveDef, getter, setter)